src/components/SUSHIControls.js

```diff
@@ -70,17 +83,48 @@ export function sliceDependency(dependencies) {
 
 export default function SUSHIControls(props) {
   const classes = useStyles();
-  const [open, setOpen] = useState(false);
+  const [openConfig, setOpenConfig] = useState(false);
+  const [openShare, setOpenShare] = useState(false);
+  const [openShareError, setOpenShareError] = useState(false);
+  const [link, setLink] = useState();
+  const [{ copied, copyButton }, setCopied] = useState({ copied: false, copyButton: 'Copy to Clipboard' });
   const [canonical, setCanonical] = useState('http://example.org');
   const [version, setVersion] = useState('1.0.0');
   const [dependencies, setDependencies] = useState('');
 
-  const handleOpen = () => {
-    setOpen(true);
+  const handleOpenConfig = () => {
+    setOpenConfig(true);
   };
 
-  const handleClose = () => {
-    setOpen(false);
+  const handleCloseConfig = () => {
+    setOpenConfig(false);
+  };
+
+  const handleOpenShare = async () => {
+    const encoded = deflateSync(props.text).toString('base64');
+    const longLink = `https://fshschool.org/FSHOnline/share/${encoded}`;
+    const bitlyLink = await generateLink(longLink);
+    if (bitlyLink.errorNeeded === true) {
+      handleOpenShareError();
+    } else {
+      const bitlySlice = bitlyLink.link.slice(15);
```

Comment:
Can you add a comment to this line reminding future-us what the 15 represent in this slice?